### ISSUE TITLE
Reconnecting to FAF forces the client to close itself.

### ISSUE THREAD

**Issue by MLGTASTICa** [FAForever/downlords-faf-client#2894]
**Describe the bug**
Reconnecting to FAF through the client after a sudden interruption forces the client to close itself, more specifically its told by the server. Also closing the FAF adapter and disallowing reconnects to game lobbies

**To Reproduce**
Get disconnected for 10-30 seconds
Use the reconnect button 
Wait for reconnect , client shortly starts "blinking" in the app tab on windows , then closes itself, if the reconnect button is pressed twice , it sends back to the menu.

**Log or error message**
Logs for the past days, containing cases of sudden disconnects and the client being told to shut down.
"[reactor-tcp-nio-2] c.f.client.remote.FafServerAccessor      : Kicked from lobby, client closing after delay" (Even if ingame , the timer for reconnecting hasn't passed)
This is my log file:
[client.log.2023-01-17.0.gz](https://github.com/FAForever/downlords-faf-client/files/10446327/client.log.2023-01-17.0.gz)
[client.log.2023-01-11.0.gz](https://github.com/FAForever/downlords-faf-client/files/10446328/client.log.2023-01-11.0.gz)
[client.log.2023-01-12.0.gz](https://github.com/FAForever/downlords-faf-client/files/10446329/client.log.2023-01-12.0.gz)
[client.log.2023-01-13.0.gz](https://github.com/FAForever/downlords-faf-client/files/10446330/client.log.2023-01-13.0.gz)

[client.log.2023-01-14.0.gz](https://github.com/FAForever/downlords-faf-client/files/10446322/client.log.2023-01-14.0.gz)
[client.log.2023-01-14.1.gz](https://github.com/FAForever/downlords-faf-client/files/10446323/client.log.2023-01-14.1.gz)
[client.log.2023-01-15.0.gz](https://github.com/FAForever/downlords-faf-client/files/10446324/client.log.2023-01-15.0.gz)
[client.log.2023-01-16.0.gz](https://github.com/FAForever/downlords-faf-client/files/10446325/client.log.2023-01-16.0.gz)

**Expected behavior**
The reconnect happens normally , client doesn't shut itself and the ICE adapter down.

**OS**
Windows 10

**Comment by Sheikah45** [comment]
These logs are hard to parse through with so many. It would be better to try and get an isolated incidence with trace logging on

**Comment by TakesTheBiscuit** [comment]
This may also be related:

```
2023-01-19 21:55:54.447  WARN 18604 --- [reactor-htt%USER%-nio-3] c.f.client.status.Stat%USER%ingService        : Could not read Stat%USER%ing from: htt%USER%s://status.faforever.com/a%USER%i

org.s%USER%ringframework.web.reactive.function.client.WebClientRes%USER%onseExce%USER%tion$NotFound: 404 Not Found from GET htt%USER%s://status.faforever.com/a%USER%i/services
	at org.s%USER%ringframework.web.reactive.function.client.WebClientRes%USER%onseExce%USER%tion.create(WebClientRes%USER%onseExce%USER%tion.java:223)
	Su%USER%%USER%ressed: reactor.core.%USER%ublisher.FluxOnAssembly$OnAssemblyExce%USER%tion: 
Error has been observed at the following site(s):
	*__check%USER%oint ? 404 from GET htt%USER%s://status.faforever.com/a%USER%i/services [DefaultWebClient]
Original Stack Trace:
		at org.s%USER%ringframework.web.reactive.function.client.WebClientRes%USER%onseExce%USER%tion.create(WebClientRes%USER%onseExce%USER%tion.java:223)
		at org.s%USER%ringframework.web.reactive.function.client.DefaultClientRes%USER%onse.lambda$createExce%USER%tion$1(DefaultClientRes%USER%onse.java:207)
		at reactor.core.%USER%ublisher.FluxMa%USER%$Ma%USER%Subscriber.onNext(FluxMa%USER%.java:106)
		at reactor.core.%USER%ublisher.FluxOnErrorReturn$ReturnSubscriber.onNext(FluxOnErrorReturn.java:162)
		at reactor.core.%USER%ublisher.FluxDefaultIfEm%USER%ty$DefaultIfEm%USER%tySubscriber.onNext(FluxDefaultIfEm%USER%ty.java:101)
		at reactor.core.%USER%ublisher.FluxMa%USER%Fuseable$Ma%USER%FuseableSubscriber.onNext(FluxMa%USER%Fuseable.java:129)
		at reactor.core.%USER%ublisher.FluxContextWrite$ContextWriteSubscriber.onNext(FluxContextWrite.java:107)
		at reactor.core.%USER%ublisher.FluxMa%USER%Fuseable$Ma%USER%FuseableConditionalSubscriber.onNext(FluxMa%USER%Fuseable.java:299)
		at reactor.core.%USER%ublisher.FluxFilterFuseable$FilterFuseableConditionalSubscriber.onNext(FluxFilterFuseable.java:337)
		at reactor.core.%USER%ublisher.O%USER%erators$MonoSubscriber.com%USER%lete(O%USER%erators.java:1816)
		at reactor.core.%USER%ublisher.MonoCollect$CollectSubscriber.onCom%USER%lete(MonoCollect.java:160)
		at reactor.core.%USER%ublisher.FluxMa%USER%$Ma%USER%Subscriber.onCom%USER%lete(FluxMa%USER%.java:144)
		at reactor.core.%USER%ublisher.Flux%USER%eek$%USER%eekSubscriber.onCom%USER%lete(Flux%USER%eek.java:260)
		at reactor.core.%USER%ublisher.FluxMa%USER%$Ma%USER%Subscriber.onCom%USER%lete(FluxMa%USER%.java:144)
```

**Comment by Sheikah45** [comment]
That is not related. Just a service that isn't up to test uptime

**Comment by TakesTheBiscuit** [comment]
[client.log.2023-01-19.0.gz](https://github.com/FAForever/downlords-faf-client/files/10462183/client.log.2023-01-19.0.gz)

Check this out - i joined a ladder at about 21:20 in the log, and around 21:50 things went wrong - i hope this helps iron down the period of time - i appreciate it might not be trace but there are clear examples of disconnects.

Happy to try and reproduce it several times; i've set my log level to trace and can try a few times tomorrow to replicate.

**Comment by Sheikah45** [comment]
Unfortunately this log was not set to trace so doesn't contain the server messages

**Comment by MLGTASTICa** [comment]
> These logs are hard to parse through with so many. It would be better to try and get an isolated incidence with trace logging on

Will do ,  it happens very frequently .

**Comment by MLGTASTICa** [comment]
> These logs are hard to parse through with so many. It would be better to try and get an isolated incidence with trace logging on

Log with Trace logging on , FAF lost connection on match start and client closed itself on reconnect.
[client.zip](https://github.com/FAForever/downlords-faf-client/files/10466375/client.zip)

**Comment by Sheikah45** [comment]
Fixed in #1754
